diff --git a/dist/esm/index.js b/dist/esm/index.js
index 4b3787d5847010784db77ec66b95d5ee1a812249..46390b149d97fd58ef9a8199fabc1342a280f4a6 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -228,7 +228,7 @@ var groupHighlightsByPage = (highlights) => highlights.reduce((acc, highlight) =
     highlight.position.boundingRect.pageNumber,
     ...highlight.position.rects.map((rect) => rect.pageNumber || 0)
   ];
-  pageNumbers.forEach((pageNumber) => {
+  new Set(pageNumbers).forEach((pageNumber) => {
     acc[pageNumber] ||= [];
     const pageSpecificHighlight = {
       ...highlight,
@@ -1235,6 +1235,8 @@ var PdfHighlighter = ({
   );
   const findControllerRef = useRef5(null);
   const viewerRef = useRef5(null);
+  const pdfScaleValueRef = useRef5(pdfScaleValue);
+  pdfScaleValueRef.current = pdfScaleValue;
   highlightsRef.current = highlights;
   childrenRef.current = children;
   useLayoutEffect2(() => {
@@ -1420,7 +1422,7 @@ var PdfHighlighter = ({
   };
   const handleScaleValue = () => {
     if (viewerRef.current) {
-      viewerRef.current.currentScaleValue = pdfScaleValue.toString();
+      viewerRef.current.currentScaleValue = pdfScaleValueRef.current.toString();
     }
   };
   const renderHighlightLayer = (highlightBindings, pageNumber, shouldRenderHighlight) => {
